cmd/virtual-worker/main.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"os"
+
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/tink/cmd/virtual-worker/cmd"
+)
+
+const (
+	serviceKey = "github.com/tinkerbell/tink"
+)
+
+// version is set at build time.
+var version = "devel"
+
+func main() {
+	logger, err := log.Init(serviceKey)
+	if err != nil {
+		panic(err)
+	}
+
+	rootCmd := cmd.NewRootCommand(version, logger)
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+	logger.Close()
```

Comment:
This should probably be defer'd after the err check to ensure Close is called and any buffered messages are flushed.

Reply:
> a couple small fixes needed but otherwise lgtm. I'd _really_ love it if #584 would land first and then your root.go could be based off of tink-worker's from then, but if not then rebasing and adapting this in #584 would be fine too.

I'm happy to wait on #584 and reuse cmd code from tink-worker

Reply:
Ah yeah I see that now. Yep lets fix it there too.

Reply:
Ah you know what, golint-ci complains that the defer won't be called with the `os.Exit(1)` and requires this to be called here.